backend/src/apiserver/storage/object_store.go

```diff
@@ -121,6 +128,88 @@ func (m *MinioObjectStore) GetFromYamlFile(o interface{}, filePath string) error
 	return nil
 }
 
+// GetSignedUrl generates a signed url for the artifact identified by artifactURI and bucketConfig.
+// The URL expires after expirySeconds. The secret contains the credentials for accessing the object
+// store for this artifact. Signed URLs are built using the "GET" method, and are only intended for
+// Artifact downloads.
+// TODO: Add support for irsa and gcs app credentials pulled from environment
+func (m *MinioObjectStore) GetSignedUrl(bucketConfig *objectstore.Config, secret *v1.Secret, expirySeconds time.Duration, artifactURI string) (string, error) {
+	s3Client, err := buildClientFromConfig(bucketConfig, secret)
+	if err != nil {
+		return "", err
+	}
+
+	prefixSplit := strings.Split(bucketConfig.Prefix, "/")
+	filename := prefixSplit[len(prefixSplit)-1]
+	reqParams := make(url.Values)
+	reqParams.Set("response-content-disposition", fmt.Sprintf("attachment; filename=\"%s\"", filename))
+
+	key, err := objectstore.ArtifactKeyFromURI(artifactURI)
+	if err != nil {
+		return "", err
+	}
+	signedUrl, err := s3Client.Presign("GET", bucketConfig.BucketName, key, expirySeconds, reqParams)
+	if err != nil {
+		return "", util.Wrap(err, "Failed to generate signed url")
+	}
+
+	return signedUrl.String(), nil
+}
+
+// GetObjectSize Retrieves the Size of the object in bytes.
+// Return zero with no error if artifact URI does not exist.
+func (m *MinioObjectStore) GetObjectSize(bucketConfig *objectstore.Config, secret *v1.Secret, artifactURI string) (int64, error) {
+	s3Client, err := buildClientFromConfig(bucketConfig, secret)
+	if err != nil {
+		return 0, err
+	}
+	key, err := objectstore.ArtifactKeyFromURI(artifactURI)
+	if err != nil {
+		return 0, err
+	}
+	objectInfo, err := s3Client.StatObject(bucketConfig.BucketName, key, minio.StatObjectOptions{})
+	if err != nil {
+		errResponse := minio.ToErrorResponse(err)
+		if errResponse.Code == "NoSuchKey" {
+			return 0, nil
+		}
+		return 0, err
+	}
+	if err != nil {
+		return 0, err
+	}
```

Comment:
`err` is always `nil`. This can be removed.

Reply:
addressed